tests/jerry/es2015/symbol.js

```diff
@@ -0,0 +1,109 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+var a = Symbol ('foo');
+var b = Symbol ('bar');
+
+assert (a !== b);
+assert (a === a);
+assert (typeof a === 'symbol')
+assert (a.toString() == 'Symbol(foo)')
+assert (Object.prototype.toString.call (a) === "[object Symbol]");
+
+var o = { c : 10, d : 20};
+o[a] = 'EnumerableSymbolProp';
+assert (o[a] === 'EnumerableSymbolProp');
+
+// Symbol properties are not listed via for in
+Object.defineProperty(o, b, { value : 'NonEnumerableSymbolProp' });
+
+var counter = 0;
+
+for (var v in o) {
+  assert (v === 'c' || v === 'd');
+  counter++;
+}
+
+assert (counter === 2);
+
+var keys = Object.keys (o);
+assert (keys.toString() === 'c,d')
+assert (keys.length === 2)
+
+var c = Symbol ('bar');
+var o2 = {};
+o2[b] = 'foo'
+o2[c] = 'bar'
+
+assert (o2[b] == 'foo')
+assert (o2[c] == 'bar')
```

Comment:
Also it would be good to have a test like:
```js
sym_a = Symbol('AA');
obj = { AA: 55 };
obj[sym_a] = 77;
```
This would check if the properties does not overwrite each other.